### ISSUE TITLE
MP Fault Tolerance annotations at the class level of a Rest Client interface are ignored

### ISSUE THREAD

**Issue by andycandy-de** [OpenLiberty/open-liberty#19826]
**Describe the bug**  
I created a `RestClient` interface and a `ResponseExceptionMapper`. I added the `@Retry` annotation to the `RestClient` interface to retry the request when a specific exception is thrown.
The exception is thrown but no retry is executed.

**Steps to Reproduce**  

RetryTestException.java
```java
public class RetryTestException extends WebApplicationException {

    public RetryTestException(Response response) {
        super("Retry", response);
    }
} 
```

RetryTestREST.java
```java
@RegisterRestClient(baseUri = "http://localhost:4567/api")
@RegisterProvider(RetryTestResponseExceptionMapper.class)
@Retry(retryOn = RetryException.class, maxRetries = 1)
public interface RetryTestREST {

    @GET
    @Path("test")
    String test() throws RetryTestException;
}
```

RetryTestResponseExceptionMapper.java
```java
public class RetryTestResponseExceptionMapper implements ResponseExceptionMapper<RetryTestException> {


    @Override
    public RetryTestException toThrowable(Response response) {
        return new RetryTestException(response);
    }

    @Override
    public boolean handles(int status, MultivaluedMap<String, Object> headers) {
        return true;
    }
}
```

Execute the `RetryTestREST.test()` method.

**Expected behavior**  
There should be a retry request.

**Diagnostic information:**  
 - Open Liberty Version 21.0.0.12
 - Affected feature(s) MP Fault Tolerance, MP Rest Client
 - Java Version: openjdk version "17.0.1" 2021-10-19 (Runtime Environment Temurin-17.0.1+12 (build 17.0.1+12) x64)

**Comment by Azquelt** [comment]
It looks like the fault tolerance annotation is not picked up when it's placed at the class level. It is picked up if the annotation is placed on the method instead.

```java
@RegisterRestClient(baseUri = "http://localhost:4567/api")
@RegisterProvider(RetryTestResponseExceptionMapper.class)
public interface RetryTestREST {

    @GET
    @Path("test")
    @Retry(retryOn = RetryException.class, maxRetries = 1)
    String test() throws RetryTestException;
}
```

This behaviour seems wrong to me but I didn't do the integration with MP Rest Client so I need to check that is isn't intentional.

**Comment by Azquelt** [comment]
I've updated the title to better describe the situation where Fault Tolerance annotations don't work.

Looking at the Rest Client spec, I think this scenario is valid, but it's not a trivial change to make it work
> MP Rest Client implementations must ensure that MP Fault Tolerance annotations on client interfaces are honored. In general, these annotations are treated as CDI interceptor bindings.

**Comment by Azquelt** [comment]
This needs two changes:
* Rest Client has some specific handling for Fault Tolerance annotations on methods, it needs to apply the same handling to annotations on the interface.
* Fault Tolerance needs to specify and consume a service to allow other features to change how it looks for annotations on the target class. Rest Client would provide an implementation of this service which can identify a rest client proxy class and provide the annotations from the corresponding interface.

**Comment by Azquelt** [comment]
@andycandy-de I have a fix in a PR for `mpRestClient-3.0`. Is that sufficient for you or do you need it backported to earlier versions as well?

`mpRestClient-3.0` is part of `microprofile-5.0` and is compatible with `mpFaultTolerance-4.0` and Jakarta EE 9 features.

**Comment by andycandy-de** [comment]
For me it's ok if it`s part of the next microprofile 5 release. I don't know how complicated a backport is but it's still an open bug in previous MP versions
